floss/language/go/extract.py

```diff
@@ -0,0 +1,431 @@
+        .text:00000000004033D2 48 89 C7                      mov     rdi, rax
+        .text:00000000004033D5 48 89 DE                      mov     rsi, rbx
+        .text:00000000004033D8 31 C0                         xor     eax, eax
+        .text:00000000004033DA 48 8D 1D C3 A1 0A 00          lea     rbx, aComparingUncom ; "comparing uncomparable type "
+        .text:00000000004033E1 E8 5A 63 04 00                call    runtime_concatstring2
+        """
+        longstring64_4 = re.compile(b"\xb9(?=(.)...........\x48\x8D.(....))", re.DOTALL)
+
+    elif pe.FILE_HEADER.Machine == pefile.MACHINE_TYPE["IMAGE_FILE_MACHINE_I386"]:
+        """
+        .data:102A78D0 E3 9A 17 10                       dd offset aString
+        .data:102A78D4 12                                db  12h
+        """
+        alignment = 0x8
+        fmt = "<II"
+
+        """
+        .text:0048CED3 75 6D                                         jnz     short loc_48CF42
+        .text:0048CED5 81 7D 00 50 61 73 73                          cmp     dword ptr [ebp+0], 73736150h
+        .text:0048CEDC 75 64                                         jnz     short loc_48CF42
+        .text:0048CEDE 66 81 7D 04 77 6F                             cmp     word ptr [ebp+4], 6F77h
+        .text:0048CEE4 75 5C                                         jnz     short loc_48CF42
+        .text:0048CEE6 80 7D 06 72                                   cmp     byte ptr [ebp+6], 72h ; 'r'
+        .text:0048CEEA 75 56                                         jnz     short loc_48CF42
+        """
+        combinedregex = re.compile(
+            b"\x81\xf9(....)|\x81\x38(....)|\x81\x7d\x00(....)|\x81\x3B(....)|\x66\x81\xf9(..)|\x66\x81\x7b\x04(..)|\x66\x81\x78\x04(..)|\x66\x81\x7d\x04(..)|\x80\x7b\x06(.)|\x80\x7d\x06(.)|\x80\xf8(.)|\x80\x78\x06(.)",
+            re.DOTALL,
+        )
+
+        """
+        .text:0048CED0 83 F8 13                                      cmp     eax, 13h
+        .text:0048CED3 75 23                                         jnz     short loc_48CEF8
+        .text:0048CED5 89 2C 24                                      mov     [esp+0B0h+var_B0], ebp
+        .text:0048CED8 8D 05 3A 49 4A 00                             lea     eax, unk_4A493A
+        .text:0048CEDE 89 44 24 04                                   mov     [esp+0B0h+var_AC], eax
+        .text:0048CEE2 C7 44 24 08 13 00 00 00                       mov     [esp+0B0h+var_A8], 13h
+        """
+        longstring32 = re.compile(b"\x83(?=.(.).....\x8D\x05(....))", re.DOTALL)
+
+        """
+        .text:00403276 8D 15 64 63 4A 00                             lea     edx, unk_4A6364
+        .text:0040327C 89 54 24 04                                   mov     [esp+1Ch+var_18], edx
+        .text:00403280 C7 44 24 08 1C 00 00 00                       mov     [esp+1Ch+var_14], 1Ch
+        """
+        longstring32_2 = re.compile(b"\x8D.(?=(....)........(.))", re.DOTALL)
```

Comment:
consider using named groups for all patterns, see https://docs.python.org/3/howto/regex.html

```
p = re.compile(r'(?P<word>\b\w+\b)')
m = p.search( '(((( Lots of punctuation )))' )
m.group('word')
'Lots'
```